packages/rum/src/getActionNameFromElement.ts

```diff
@@ -0,0 +1,176 @@
+      return label && getTextualContent(label)
+    }
+  },
+  // INPUT button (and associated) value
+  (element) => {
+    if (element.nodeName === 'INPUT') {
+      const input = element as HTMLInputElement
+      const type = input.getAttribute('type')
+      if (type === 'button' || type === 'submit' || type === 'reset') {
+        return input.value
+      }
+    }
+  },
+  // BUTTON, LABEL or button-like element text
+  (element) => {
+    if (element.nodeName === 'BUTTON' || element.nodeName === 'LABEL' || element.getAttribute('role') === 'button') {
+      return getTextualContent(element)
+    }
+  },
+  // aria-label attribute value
+  (element) => element.getAttribute('aria-label'),
+  // associated element text designated by the aria-labelledby attribute
+  (element) => {
+    const labelledByAttribute = element.getAttribute('aria-labelledby')
+    if (labelledByAttribute) {
+      return labelledByAttribute
+        .split(/\s+/)
+        .map((id) => getElementById(element, id))
+        .filter((label): label is HTMLElement => Boolean(label))
+        .map(getTextualContent)
+        .join(' ')
+    }
+  },
+  // alt attribute value
+  (element) => element.getAttribute('alt'),
+  // name attribute value
+  (element) => element.getAttribute('name'),
+  // title attribute value
+  (element) => element.getAttribute('title'),
+  // placeholder attribute value
+  (element) => element.getAttribute('placeholder'),
+  // SELECT first OPTION text
+  (element) => {
+    if ('options' in element && element.options.length > 0) {
+      return getTextualContent(element.options[0])
+    }
+  },
+]
+
+const fallbackGetters: NameGetter[] = [
+  // element text
+  (element) => {
+    return getTextualContent(element)
+  },
+]
+
+function getActionNameFromElementForGetters(targetElement: Element, getters: NameGetter[]) {
+  let element: Element | null = targetElement
+  let recursionCounter = 0
+  while (
+    recursionCounter <= 10 &&
+    element &&
+    element.nodeName !== 'BODY' &&
+    element.nodeName !== 'HTML' &&
+    element.nodeName !== 'HEAD'
+  ) {
+    for (const getter of getters) {
+      const name = getter(element)
+      if (typeof name === 'string') {
+        const trimedName = name.trim()
+        if (trimedName) {
+          return truncate(normalizeWhitespace(trimedName))
+        }
+      }
+    }
+    // Don't consider parents of FORM elements
+    if (element.nodeName === 'FORM') {
+      break
+    }
+    element = element.parentElement
+    recursionCounter += 1
+  }
+}
+
+function normalizeWhitespace(s: string) {
+  return s.replace(/\s+/g, ' ')
+}
+
+function truncate(s: string) {
+  return s.length > 100 ? `${s.slice(0, 100)} [...]` : s
+}
+
+function getElementById(refElement: Element, id: string) {
+  // tslint:disable-next-line: no-null-keyword
+  return refElement.ownerDocument ? refElement.ownerDocument.getElementById(id) : null
```

Comment:
what is the reasoning for using `elt.ownerDocument` instead of `document`?

Reply:
Tests are using iframes to emulate a full page context. The `document.getElementeById` doesn't work in this case. Also it may be useful sometime if we want to support iframes in some ways.

Reply:
iframe support could be quite challenging 😄 
since it is not the go to API for that, I think a small comment could be helpful